src/shims/generic_win_stubs.c

```diff
@@ -1,24 +1,40 @@
 #include "internal.h"
 
-/*
- * This file contains stubbed out functions we are using during
- * the initial Windows port.  When the port is complete, this file
- * should be empty (and thus removed).
- */
+typedef void (WINAPI *_precise_time_fn_t)(PULONGLONG);
 
-void
-_dispatch_runloop_queue_dispose(dispatch_queue_t dq DISPATCH_UNUSED,
-		bool *allow_free DISPATCH_UNUSED)
+DISPATCH_STATIC_GLOBAL(dispatch_once_t _dispatch_precise_time_pred);
+DISPATCH_STATIC_GLOBAL(_precise_time_fn_t _dispatch_QueryInterruptTimePrecise_ptr);
+DISPATCH_STATIC_GLOBAL(_precise_time_fn_t _dispatch_QueryUnbiasedInterruptTimePrecise_ptr);
+
+static void
+_dispatch_init_precise_time(void *context DISPATCH_UNUSED)
 {
-	WIN_PORT_ERROR();
+	HMODULE kernelbase = LoadLibraryW(L"KernelBase.dll");
+	if (!kernelbase) {
+		DISPATCH_INTERNAL_CRASH(0, "failed to load KernelBase.dll");
+	}
+	_dispatch_QueryInterruptTimePrecise_ptr = (_precise_time_fn_t)
+			GetProcAddress(kernelbase, "QueryInterruptTimePrecise");
+	_dispatch_QueryUnbiasedInterruptTimePrecise_ptr = (_precise_time_fn_t)
+			GetProcAddress(kernelbase, "QueryUnbiasedInterruptTimePrecise");
+	if (!_dispatch_QueryInterruptTimePrecise_ptr) {
+		DISPATCH_INTERNAL_CRASH(0, "could not locate QueryInterruptTimePrecise");
+	}
+	if (!_dispatch_QueryUnbiasedInterruptTimePrecise_ptr) {
+		DISPATCH_INTERNAL_CRASH(0, "could not locate QueryUnbiasedInterruptTimePrecise");
+	}
 }
 
 void
-_dispatch_runloop_queue_xref_dispose(dispatch_queue_t dq DISPATCH_UNUSED)
```

Comment:
Is this not used anymore?

Reply:
This file had been removed from the build but not deleted. We have to remove these stubs or else we get duplicate symbol errors.

Reply:
I think maintaining the existing structure is preferable since it mirrors the other targets.